packages/core/src/labels.ts

```diff
@@ -15,19 +18,30 @@ export type Labels = Record<string, string | number | boolean>;
  * removed. See {@link setLabels} if you want to replace the existing labels
  * entirely.
  */
-export function withLabels<T>(operation: Operation<T>, labels: Labels): Operation<T> {
-  if(operation) {
-    operation.labels = { ...(operation.labels || {}), ...labels };
+export function withLabels<T>(operation: Operation<T>, labels: Labels): OperationObject<T> {
```

Comment:
`withLabels` start returning `OperationObject` instead of just `Operation` is it ok?

Reply:
As discussed, let's keep the mutative behavior. In `v3` we will have `withLabels` just return an OperationObject. This will also allow us to get rid of the `Labelled` interface entirely. But for now, we have to keep it as is.

Of course, since OperationObject is a new api and does not have existing behavior, we can return a new version of it. Although there is little point in passing an object to `withLabels` since you could just create the delegation yourself. In fact, we'll probably deprecate `withLabels` entirely in `v3` and remove it in `v4`.